src/analyzing/AST_to_IL.ml

```diff
@@ -1452,6 +1470,21 @@ and stmt_aux env st =
   | G.OtherStmtWithStmt (G.OSWS_Block _, [ G.E objorig ], stmt1) ->
       let ss, _TODO_obj = expr_with_pre_stmts env objorig in
       ss @ stmt env stmt1
+  (* Ruby: begin ... rescue ... else ... ensure ... *)
+  | G.OtherStmtWithStmt
```

Comment:
Usually it's better to keep the AST_to_IL.ml as simple as possible and as general as possible. So here it would probably be better to adapt ruby_to_generic.ml to generate a better AST so that the existing code in AST_to_IL.ml would handle it for free.
I'm not super familiar with the semantic of begin/try/else/rescue in Ruby, so that's why I hacked the current translation, because I didn't have the time to make it better, but if you now fully understand the semantic of it, then just improve ruby_to_generic.ml to generate the right Try/Catch/Finally construct directly,  possibly extending the generic AST to make the current try/catch construct more general.

Those OtherStmtWithStmt are used as a last resort when we don't have the time to fully understand the semantic and we still want some CFG for it and some support for it, but in the long term we should never use those last-resort cases.

Reply:
Could we then try adding

```ocaml
type try_else = tok * stmt
...
| Try of tok * stmt * catch list * try_else option * finally option 
```

?

Or do we need something else?

Reply:
I'd like to try adding this and will update on findings.

Reply:
This is direction I've been taking. Seems like things are looking good. Resolving this comment.